lib/fizzy/capi.cpp

```diff
@@ -343,15 +343,36 @@ inline FizzyExportDescription wrap(const fizzy::Export& exp) noexcept
 }  // namespace
 
 extern "C" {
-bool fizzy_validate(const uint8_t* wasm_binary, size_t wasm_binary_size)
+bool fizzy_validate(const uint8_t* wasm_binary, size_t wasm_binary_size, FizzyError* error)
 {
     try
     {
         fizzy::parse({wasm_binary, wasm_binary_size});
+        // TODO set_success helper
+        if (error != nullptr)
+        {
+            error->code = FIZZY_SUCCESS;
+            error->message[0] = '\0';
+        }
         return true;
     }
-    catch (...)
+    catch (const fizzy::exception& e)
+    {
+        // TODO set_error helper
+        if (error != nullptr)
+        {
+            error->code = e.code;
+            snprintf(error->message, std::size(error->message), "%s", e.what());
```

Comment:
Wouldn't `strlcpy(error->message, e.what(), std::size(error->message))` be safer? (Assuming `e.what()` is a `char*` and `std::size` is equivalent to `sizeof(error->message)`)

See: https://linux.die.net/man/3/strlcpy

I think evmc already uses this in the loader and has an implementation for Windows?

Reply:
But @chfast may be right in the sense, that this complicates matters a bit, e.g. in case of truncation we may want to have a trailing `...` which makes this worse (at least the output buffer size is set by us, within `FizzyError`, but still).

Reply:
I used here `snprintf`, because it's standard. 
evmc has its own implementation of `strlcpy` that truncates to empty string in case it doesn't fit https://github.com/ethereum/evmc/blob/c39de40f025673575412e71afdf9f72f8c6314ba/lib/loader/loader.c#L52
and also uses `vsnprintf` for error messages
https://github.com/ethereum/evmc/blob/c39de40f025673575412e71afdf9f72f8c6314ba/lib/loader/loader.c#L85